DOC: Update np.unique_all example to demonstrate namedtuple output

t-tasin opened the pull request:
DOC: **update** `np.unique_all` example to show **namedtuple** output

Replaced `np.unique` example with `np.unique_all` to demonstrate the 
namedtuple result format. Clarified output fields: `values`, `indices`, 
`inverse_indices`, and `counts`. This change enhances documentation clarity 
by showing the actual output of `np.unique_all`.

DOC: **update** `np.unique_counts` example to show **namedtuple** output

Replaced `np.unique` example with `np.unique_counts` to demonstrate the 
namedtuple result format. Clarified output fields: `values` and `counts`. 
This change enhances documentation clarity by showing the actual 
output of `np.unique_counts`.

DOC: **update** `np.unique_inverse` example to show **namedtuple** output

Replaced `np.unique` example with `np.unique_inverse` to demonstrate the 
namedtuple result format. Clarified output fields: `values` and 
`inverse_indices`. This change enhances documentation clarity by showing 
the actual output of np.unique_inverse.


DOC: **update** `np.unique_values` example to demonstrate the correct function

Replaced `np.unique` example with `np.unique_values` to show the correct 
output of the function, returning only the unique elements of the input 
array. This change clarifies the intended behavior of np.unique_values.


Closes #27214.

Commit:
DOC: Update np.unique_all example to demonstrate namedtuple output

Commit:
DOC: update np.unique_counts example to show namedtuple output

Commit:
DOC: update np.unique_inverse example to show namedtuple output

Commit:
DOC: update np.unique_values example to demonstrate the correct function

t-tasin commented:
Hi, I have fixed this issue by updating the documentation for `np.unique_all`, `np.unique_counts`, `np.unique_inverse`, and `np.unique_values` to show examples using the respective functions, as requested. The changes demonstrate the correct namedtuple outputs (where applicable) and clarify the function behavior. 

Please review the PR and let me know if further adjustments are needed. Thanks!

jpivarski reviewed (commented):
I don't have permissions to approve this PR, but I would approve it: it completely addresses the issue. Thanks!

rossbar reviewed (changes requested):
The content changes look good but there are some docstring re-org changes that need to be backed out. Thanks @t-tasin and @jpivarski

rossbar commented on numpy/lib/_arraysetops_impl.py:
```diff
@@ -414,12 +414,15 @@ def unique_all(x):
+    This function is an Array API compatible alternative to `np.unique`, but
+    returns a namedtuple for easier access to each output.
 
+    Example
```
The naming (especially) and the ordering of the docstring sections is important and is specified by the [numpydoc style guide](https://numpydoc.readthedocs.io/en/latest/format.html).

The sections must be named `Examples` and preferably would be moved back to the original location in the docstring.

t-tasin replied:
Hey @rossbar , I've made the requested changes to follow the NumPy docstring style guide, renaming 'Example' to 'Examples' and restoring the original order of sections. Please review the updated PR. Thanks!

Commit:
DOC: docstring style updated according to request

rossbar commented on numpy/lib/_arraysetops_impl.py:
```diff
@@ -412,14 +412,10 @@ def _unique_all_dispatcher(x, /):
-    >>> np.unique(x, return_index=True, return_inverse=True,
-    ...           return_counts=True, equal_nan=False)
-    (array([1, 2]), array([0, 2]), array([0, 0, 1]), array([2, 1]))
+    This function is an Array API compatible alternative to `np.unique`, but
+    returns a namedtuple for easier access to each output.
```
I think this removes too much info - the idea here is that the `unique_*` functions behave like `unique` with the given subset of keyword arguments changed. Those changes include `equal_nan`, which differs than the default value of `np.unique` as well.

I'd propose to leave the info here, but remove the execution bit:

```rst
    This function is an Array API compatible alternative to::
    
        np.unique(x, return_index=True, return_inverse=True,
                  return_counts=True, equal_nan=False)

    but returns a namedtuple for easier access to each output.
```

rossbar commented on numpy/lib/_arraysetops_impl.py:
```diff
@@ -430,7 +426,6 @@ def unique_all(x):
     -------
     out : namedtuple
         The result containing:
-
```
This line break is necessary for the bulleted list formatting and should be added back.

rossbar commented on numpy/lib/_arraysetops_impl.py:
```diff
@@ -443,13 +438,10 @@ def unique_all(x):
 
     Examples
     --------
-    >>> import numpy as np
```
The `>>> import numpy as np` statements should not be removed. These were recently added to all the docstrings and should remain there (see e.g. #26814).

rossbar commented on numpy/lib/_arraysetops_impl.py:
```diff
@@ -443,13 +438,10 @@ def unique_all(x):
-
+    >>> x = np.array([1, 1, 2])
+    >>> np.unique_all(x)
+    UniqueAllResult(values=array([1, 2]), indices=array([0, 2]),
+                    inverse_indices=array([0, 0, 1]), counts=array([2, 1]))
```
Unforunately the spacing here matters - this is failing CI because `doctest` requires that the output from the call be character-identical with the REPL output.

Unfortunately again, fixing this is likely to cause a linter to complain due to the inherently long line length. That's okay though, passing doctests is more important.

Alternatively, the example could be recast to highlight the advantages of namedtuple without dumping all of the info on one line. For example:

```python
>>> x = [1, 1, 2]
>>> uniq = np.unique_all(x)
>>> uniq.values
array([1, 2])
>>> uniq.indices
array([0, 2])
```

... etc.

Commit:
DOC: Address code review feedback for np.unique_* functions

t-tasin commented:
@rossbar , Hi, I've made the requested updates, removed `import numpy as np` from examples, fixed indentation, and added necessary blank lines. All tests are now passing. Please review. Thanks!

rossbar committed:
Minor formatting nits.

rossbar reviewed (approved):
LGTM, thanks @t-tasin !
